DSCResources/MSFT_xNetBIOS/MSFT_xNetBIOS.psm1

```diff
@@ -0,0 +1,184 @@
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    $Netadapterparams = @{
+        ClassName = 'Win32_NetworkAdapter'
+        Filter = 'NetConnectionID="{0}"' -f $InterfaceAlias
+    }
+    $NetAdapterConfig = Get-CimInstance @Netadapterparams -ErrorAction Stop |
+            Get-CimAssociatedInstance `
+                -ResultClassName Win32_NetworkAdapterConfiguration `
+                -ErrorAction Stop
+
+    if ($Setting -eq [NETBIOSSetting]::Default) 
+    {
+        Write-Verbose -Message $LocalizedData.ResetToDefaut
+        #If DHCP is not enabled, settcpipnetbios CIM Method won't take 0 so overwrite registry entry instead.
+        $RegParam = @{
+            Path = "HKLM:\SYSTEM\CurrentControlSet\Services\NetBT\Parameters\Interfaces\Tcpip_$($NetAdapterConfig.SettingID)"
+            Name = 'NetbiosOptions'
+            Value = 0
+        }
+        $null = Set-ItemProperty @RegParam
+    }
+    else
+    {
+        Write-Verbose -Message ($LocalizedData.SetNetBIOS -f $Setting)
+        $null = $NetAdapterConfig | 
+            Invoke-CimMethod -MethodName SetTcpipNetbios -ErrorAction Stop -Arguments @{
+                TcpipNetbiosOptions = [uint32][NETBIOSSetting]::$Setting.value__
+            }
+    }
+}
+
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    $NIC = Get-CimInstance `
+        -ClassName Win32_NetworkAdapter `
+        -Filter "NetConnectionID=`"$InterfaceAlias`""
+    if ($Null -ne $NIC)
+    {
+        Write-Verbose -Message ($LocalizedData.InterfaceDetected -f $InterfaceAlias,$NIC.InterfaceIndex)
+    }
+    else
+    {
+        $ErrorParam = @{
+            ErrorId = 'NICNotFound'
+            ErrorMessage = ($LocalizedData.NICNotFound -f $InterfaceAlias)
+            ErrorCategory = 'ObjectNotFound'
+            ErrorAction = 'Stop'
+        }
+        New-TerminatingError @ErrorParam
+    }
+
+    $NICConfig = $NIC | Get-CimAssociatedInstance -ResultClassName Win32_NetworkAdapterConfiguration
+    
+    Write-Verbose -Message ($LocalizedData.CurrentNetBiosSetting -f [NETBIOSSetting].GetEnumValues()[$NICConfig.TcpipNetbiosOptions])
+
+    $DesiredSetting = ([NETBIOSSetting]::$($Setting)).value__
+    Write-Verbose -Message ($LocalizedData.DesiredSetting -f $Setting)
+
+    if ($NICConfig.TcpipNetbiosOptions -eq $DesiredSetting) 
+    {
+        Write-Verbose -Message $LocalizedData.InDesiredState
+        return $true
+    }
+    else 
+    {
+        Write-Verbose -Message $LocalizedData.NotInDesiredState
+        return $false
+    }
+}
+
+#region helper functions
+function New-TerminatingError
```

Comment:
Agreed. I use VS code and it moans about the Validate verb, but only seems to moan about Verb-Noun, if you leave the hyphen out, ie VerbNoun, it seems to not moan?! They've not exported anyway..

Perhaps the Validate functions should be called Assert instead as they throw terminating errors?

In my defence, they were called that before I got involved ;) Obviously, if they're exported from a common module - rather than dot-sourced - the functions will need to well named.